src/agent/adu_core_interface/src/adu_core_interface.c

```diff
@@ -613,6 +608,21 @@ static _Bool set_workflow_properties(
     return succeeded;
 }
 
+static STRING_HANDLE construct_extended_result_codes_str(ADUC_WorkflowHandle handle, ADUC_Result rootResult)
+{
+    STRING_HANDLE root_result_erc_str =
+        ADUC_ReportingUtils_CreateReportingErcHexStr(rootResult.ExtendedResultCode, true /* is_first */);
+    STRING_HANDLE extra_ercs_str = workflow_get_extra_ercs(handle);
```

Comment:
Check whether root_result_erc_str == null here, and bail early to avail unnecessary work?

Reply:
That is already checked on line 616 in 3rd and 4th terms of "and" conditional expr:
 if (extra_ercs_str != NULL && STRING_length(extra_ercs_str) > 0 _**&& root_result_erc_str != NULL
        && STRING_length(root_result_erc_str) > 0**_)